docs/pages/reference/api/architecture.mdx

```diff
@@ -0,0 +1,98 @@
+action on the `role` resource. You should create a user and role with the minimum permissions required.
+
+```bash
+# Copy and Paste the below and run on the Teleport Auth server.
+cat > api-role.yaml <<EOF
+kind: role
+metadata:
+  name: api-role
+spec:
+  allow:
+    rules:
+      - resources: ['role']
+        verbs: ['read']
+  deny:
+    node_labels:
+      '*': '*'
+version: v3
+EOF
+# Create role
+tctl create -f api-role.yaml
+# Add user and login via web proxy
+tctl users add api-user --roles=api-role
+```
+
+See our [roles](../../access-controls/reference.mdx#roles) documentation for more details
+on role based access control.
+
+## Credentials
+
+The Teleport Go client uses Credentials in order to gather and hold TLS certificates, connect 
+to proxy servers over SSH, and perform some other actions.
+
+Credentials are created by using Credential loaders, which gather certificates and data 
+generated by [Teleport CLIs](../../cli-docs.mdx).
+
+Since there are several Credential loaders to choose from with distinct benefits, here's a quick breakdown:
+ - Profile Credentials are the easiest to get started with. All you have to do is login
+   on your device with `tsh login`. Your Teleport proxy address and credentials will
+   automatically be located and used.
+ - IdentityFile Credentials are the most well rounded in terms of usability, functionality,
+   and customizability. Identity files can be generated through `tsh login` or `tctl auth sign`,
+   making them ideal for both long lived proxy and auth server connections.
+ - Key Pair Credentials have a much simpler implementation than the first two Credentials listed,
+   and may feel more familiar. These are good for authenticating client's hosted directly on the auth server.
+ - TLS Credentials leave everything up to the client user. This is mostly used internally, but 
+   some advanced users may find it useful.
+
+Here are some more specific details to differentiate them by:
+
+| Type | Profile Credentials | Identity Credentials | Key Pair Credentials | TLS Credentials |
+| - | - | - | - | - | 
+| Ease of use | easy | easy | med | hard | 
+| Supports long lived certificates | yes, but must be configured on server side | yes | yes | yes | 
+| Supports proxy connections | yes | yes (6.1+) | no | no | 
+| Automatic Proxy Address discovery | yes | no | no | no | 
+| CLI used | tsh | tctl/tsh | tctl | - | 
+| Available in | 6.1+ | 6.0+ | 6.0+ | 6.0+ | 
+
+See the [Credentials type](https://pkg.go.dev/github.com/gravitational/teleport/api/client#Credentials)
+on pkg.go.dev for more information and examples for Credentials and Credential Loaders.
+
+## Client Connection
+
+The API client makes requests through an open connection to the Teleport Auth server.
+This connection can be made directly or remotely.
+
+Direct Auth server connections can only be made if the client is running locally to the server.
```

Comment:
Not quite sure what do you mean here, client can issue direct mTLS connections even if it's running on a different machine, but has a networking connection to the servers HTTPS address.